src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -639,13 +654,13 @@ void CascadiaSettings::_LayerOrCreateProfile(const Json::Value& profileJson)
         // `source`. Dynamic profiles _must_ be layered on an existing profile.
         if (!Profile::IsDynamicProfileObject(profileJson))
         {
-            auto profile = winrt::make_self<Profile>();
+            auto profile{ winrt::make_self<Profile>() };
 
             // GH#2325: If we have a set of default profile settings, apply them here.
             // We _won't_ have these settings yet for defaults, dynamic profiles.
             if (_userDefaultProfileSettings)
             {
-                profile->LayerJson(_userDefaultProfileSettings);
+                profile = _userDefaultProfileSettings->CreateChild();

```

Comment:
careful - this will create a useless profile and then delete it every time

you probably want `InsertParent(0`